src/Elasticsearch.Extensions.Logging/ElasticsearchLogger.cs

```diff
@@ -129,9 +129,18 @@ private void AddTracing(LogEvent logEvent)
 
 			if (activity != null)
 			{
+				// Older version of Activity does not have TraceId, SpanId fields, but we can infer them.
+
 				// Unique identifier of the trace.
 				// A trace groups multiple events like transactions that belong together. For example, a user request handled by multiple inter-connected services.
 				logEvent.Trace = new Trace() { Id = activity.RootId };
+
+				// Use field span.id forward compatible with ECS 1.6
+				var spanId = ExtractW3cSpanIdFromActivityId(activity.Id);
```

Comment:
I think we could check if the `Activity` is not null and `activity.IdFormat == ActivityIdFormat.W3C`, and pull the trace id and span id from `activity.TraceId.ToHexString()` and `activity.SpanId.ToHexString()`, respectively. This would remove the need to do the W3C Trace Context parsing. If there is a desire to also handle the older hierarchical format, that could be handled in a separate if/else branch

Reply:
Activity.IdFormat, Activity.Span, etc are all netcore 3.0 features, and not available in netstandard2.0; hence the clunky pulling Activity.Id and seeing if the string matches the expected format (because the library is being used in a 3.0+ application).

We could do multi-targeting for the library packages, and then conditional compilation #if version >= 3.0 then grab SpanId, but I suggest maybe as a separate pull request.

Reply:
We're [using them in `Elastic.Apm`](https://github.com/elastic/apm-agent-dotnet/blob/c061236367df7b12a16e8e75d9d0abe12db34dfa/src/Elastic.Apm/Model/Transaction.cs#L183), which [targets netstandard2.0](https://github.com/elastic/apm-agent-dotnet/blob/c061236367df7b12a16e8e75d9d0abe12db34dfa/src/Elastic.Apm/Elastic.Apm.csproj#L3). Am I misunderstanding?

Reply:
Okay, so Elasticsearch.Net/7.6.1 has a dependency on System.Diagnostics.DiagnosticSource/5.0.0; which from looking is not actually part of netstandard proper, just a library built on top of it.

The documentation says you need Core 3.0 or above, https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.activity.idformat

But that's not actually correct, all you need is a nuget reference to System.Diagnostics.DiagnosticSource, which runs on top of netstandard 1.1 or above, https://www.nuget.org/packages/System.Diagnostics.DiagnosticSource/5.0.0

It seems to all compile.

Reply:
Updated to use Activity.Trace/.Span, if W3C format, otherwise fallback (similar to older fallback to TraceContext.Activity)